### PR TITLE
docs(venv-manager): document unsafe-best-match index strategy trade-off

### LINKED ISSUE #262: docs(venv-manager): document unsafe-best-match index strategy trade-off
## Context

The venv manager uses `--index-strategy unsafe-best-match` when invoking
`uv` for session venv installs. This is a deliberate trade-off: Galaxy
Proxy uses collection-namespaced package names (e.g.
`ansible-collection-community-vmware`) that do not exist on PyPI, so
dependency confusion risk is low in practice. The alternative
(`first-match`) causes transitive deps to fail resolution when they
exist only on PyPI.

However, this strategy is not documented and operators cannot disable it.

Flagged during review of PR #260:
https://github.com/ansible/apme/pull/260#discussion_r3049146499

Contributor acknowledged as a deliberate trade-off that needs documentation.

## Proposal

- Document the trade-off in deployment/operator docs
- Consider adding an env var or config option to override the index
  strategy for environments with stricter dependency policies

### PR DESCRIPTION
**Pull request by djdanielsson**
## Summary

- Add a new section to `docs/guides/DEPLOYMENT.md` explaining why the venv manager defaults to `--index-strategy unsafe-best-match` when invoking `uv pip install`
- Documents the dependency-confusion risk assessment and the `APME_UV_INDEX_STRATEGY` override for stricter environments
- Add runtime support for `APME_UV_INDEX_STRATEGY` env var to override the default index strategy in `_run_pip_install`, with `.strip()`/empty-string guard to fall back to `unsafe-best-match`
- Add unit tests for the env var override (default, custom, whitespace fallback)

## Test plan

- [x] `tox -e lint` passes
- [x] `tox -e unit -- -k TestRunPipInstallIndexStrategy --no-cov` — 3 tests pass (default strategy, custom override, whitespace fallback)
- [x] Docs wording reflects `unsafe-best-match` as default, not unconditional

Fixes #262

Made with [Cursor](https://cursor.com)

### CODE REVIEW

File: docs/guides/DEPLOYMENT.md [hunk 1]
```diff
@@ -227,6 +227,31 @@ Collections are installed into the venv's `site-packages/ansible_collections/` d
...
+**Risk assessment:** Because the Galaxy Proxy acts as a `--extra-index-url`
+alongside PyPI, `unsafe-best-match` picks the highest version across both
+indexes.  In theory an attacker who registered matching package names on PyPI
+could hijack resolution.  In practice the collection-namespaced names are not
+on PyPI, and the transitive deps (standard Python packages) resolve correctly
+from PyPI.
+
+If your environment has stricter dependency policies (e.g. air-gapped or
+internal registries), you can override the strategy by setting the
+`UV_INDEX_STRATEGY` environment variable in the Primary container.
```
[the remark this answers is not in the document]
Comment (djdanielsson) [reply docs/guides/DEPLOYMENT.md]: Good catch — the CLI flag `--index-strategy` takes precedence over `UV_INDEX_STRATEGY`. Fixed by: (1) adding `APME_UV_INDEX_STRATEGY` env var support in the venv manager so it reads the strategy at runtime, and (2) updating the docs to accurately explain that `UV_INDEX_STRATEGY` won't work and pointing to the correct override mechanism. Fixed in e1b61dd.

File: src/apme_engine/venv_manager/session.py [hunk 2]
```diff
@@ -187,7 +187,7 @@ def _run_pip_install(
             "--extra-index-url",
             simple_url,
             "--index-strategy",
-            "unsafe-best-match",
+            os.environ.get("APME_UV_INDEX_STRATEGY", "unsafe-best-match"),
```
[the remark this answers is not in the document]
Comment (djdanielsson) [reply src/apme_engine/venv_manager/session.py]: Good catch — added `.strip()` and empty-string fallback so whitespace or empty `APME_UV_INDEX_STRATEGY` values fall back to `unsafe-best-match` instead of passing an invalid value to uv. Also added a unit test (`test_empty_env_var_falls_back_to_default`) covering this case. Fixed in 2b869a9.

File: docs/guides/DEPLOYMENT.md [hunk 3]
```diff
@@ -227,6 +227,34 @@ Collections are installed into the venv's `site-packages/ansible_collections/` d
 
 The Ansible validator requires a `venv_path` from the Primary orchestrator. If none is provided (e.g., standalone testing without Primary), the validator returns an infrastructure error and skips validation.
 
+### Galaxy Proxy index strategy (`unsafe-best-match`)
+
+When the venv manager installs collections into session-scoped venvs via
+`uv pip install`, it uses `--index-strategy unsafe-best-match`.  This is a
+deliberate trade-off:
```
[the remark this answers is not in the document]
Comment (djdanielsson) [reply docs/guides/DEPLOYMENT.md]: Updated the docs to say the venv manager "defaults to" `unsafe-best-match` rather than "uses" it, making it clear the strategy is overridable via `APME_UV_INDEX_STRATEGY`. Fixed in 2b869a9.

File: src/apme_engine/venv_manager/session.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (djdanielsson) [reply src/apme_engine/venv_manager/session.py]: Correct — the PR does include a runtime code change (`_run_pip_install` now reads `APME_UV_INDEX_STRATEGY`). Updated the PR description to accurately reflect both documentation and code changes, and added unit tests. Fixed in 2b869a9.

File: src/apme_engine/venv_manager/session.py [hunk 4]
```diff
@@ -187,7 +187,7 @@ def _run_pip_install(
...
         ]
```
[the remark this answers is not in the document]
Comment (djdanielsson) [reply src/apme_engine/venv_manager/session.py:191]: Added three unit tests in `TestRunPipInstallIndexStrategy`: `test_default_strategy` (env var unset → unsafe-best-match), `test_custom_strategy` (env var set → first-match), and `test_empty_env_var_falls_back_to_default` (whitespace → falls back to default). Fixed in 2b869a9.
